### ISSUE TITLE
linter: eslint/no-lone-blocks errors on empty catch block

### ISSUE THREAD

**Issue by drvn-mr** [oxc-project/oxc#8515]
### What version of Oxlint are you using?

0.15.6

### What command did you run?

`oxlint -c .oxlintrc.json test.js`

### What does your `.oxlint.json` config file look like?

```json5
{
  "rules": {
    "no-lone-blocks": "error"
  }
}
```

### What happened?

Cannot have an empty catch block, even with a comment.
This is not reported as an error by ESLint.

Code:
```javascript
try {
  let a = 1;
  console.log(a);
} catch {}
```

```console
❯ oxlint -c .oxlintrc.json test.js

  × eslint(no-lone-blocks): Block is unnecessary.
   ╭─[test.js:4:9]
 3 │   console.log(a);
 4 │ } catch {}
   ·         ──
   ╰────

Finished in 10ms on 1 file with 98 rules using 8 threads.
Found 0 warnings and 1 error.
```
